core/utils/http.go

```diff
@@ -108,13 +108,16 @@ func withRetry(
 		case *HTTPResponseTooLargeError:
 			return responseBody, statusCode, err
 		}
-		// Sleep and retry.
+		// Sleep and retry, unless the parent context is
```

Comment:
In the case that our Observe context times out, we want to immediately give up on retrying the http requests. I think that in general we'd want that behaviour as well.